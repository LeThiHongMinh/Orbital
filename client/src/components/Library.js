import React, { useState } from 'react';
import LibDisplay from './LibDisplay';
import SearchBar from './SearchBarLib';
import FileList from './FileList';
import Nav from './Nav';

const Library = () => {
  const [isFormVisible, setFormVisible] = useState(false);

  const handleShowForm = () => {
    setFormVisible(true);
  };

  const handleCloseForm = () => {
    setFormVisible(false);
  };

  return (
    <div className=" mx-auto p-4 bg-red-100">
      <div className="flex justify-between items-center mb-6">
        <Nav />
        <h1 className="text-3xl font-bold items-center justify-center text-red-800"></h1>
        <button
          onClick={handleShowForm}
          className="bg-red-600 hover:bg-red-700 mt-20 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
        >
          Upload Notes
        </button>
      </div>
      <SearchBar />
      <div className="flex justify-center">
        <FileList />
      </div>
      <LibDisplay visible={isFormVisible} onClose={handleCloseForm} />
    </div>
  );
};

export default Library;
