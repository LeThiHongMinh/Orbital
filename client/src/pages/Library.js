import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import Layout from '../components/layout'; // Adjust the path as necessary
import LibDisplay from '../components/LibDisplay';
import SearchBar from '../components/SearchBarLib';
import FileList from '../components/FileList';

const Library = () => {
  const [isFormVisible, setFormVisible] = useState(false);

  const isDarkMode = useSelector((state) => state.ui.isDarkMode); // Access dark mode state from Redux store

  const handleShowForm = () => {
    setFormVisible(true);
  };

  const handleCloseForm = () => {
    setFormVisible(false);
  };

  return (
    <Layout>
      <div className={`mx-auto p-4 ${isDarkMode ? 'bg-gray-900' : 'bg-red-100'}`}>
        <div className={`max-w-4xl mx-auto ${isDarkMode ? 'text-white' : 'text-black'}`}>
          <div className="flex justify-between items-center mb-6">
            <h1 className={`text-3xl font-bold ${isDarkMode ? 'text-white' : 'text-red-800'}`}>Library</h1>
            <button
              onClick={handleShowForm}
              className={`mt-20 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline ${isDarkMode ? 'bg-[#bb86fc] hover:bg-[#9e63f7]' : 'bg-red-600 hover:bg-red-700'}`}
            >
              Upload Notes
            </button>
          </div>
          <SearchBar />
          <div className="flex justify-center">
            <FileList />
          </div>
          <div className="mt-8">
            <LibDisplay visible={isFormVisible} onClose={handleCloseForm} />
          </div>
        </div>
      </div>
    </Layout>
  );
};

export default Library;
