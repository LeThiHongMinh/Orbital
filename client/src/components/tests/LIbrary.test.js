import * as React from 'react'
import Library from '../Library'
import renderer from 'react-test-renderer'

jest.useFakeTimers()

it('renders correctly', () => {
  const tree = renderer.create(<Library />)
  expect(tree).toMatchSnapshot()
})