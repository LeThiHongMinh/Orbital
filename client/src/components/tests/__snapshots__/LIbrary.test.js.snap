// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container mx-auto p-4"
>
  <div
    className="flex justify-between items-center mb-6"
  >
    <h1
      className="text-3xl font-bold"
    >
      Library
    </h1>
    <button
      className="bg-indigo-600 hover:bg-indigo-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
      onClick={[Function]}
    >
      Upload Notes
    </button>
  </div>
  <div
    className="max-w-lg mx-auto mt-10 p-6 bg-white rounded-lg shadow-md"
  >
    <h2
      className="text-2xl font-bold mb-6 text-center"
    >
      Search Books
    </h2>
    <div
      className="mb-4"
    >
      <input
        className="shadow appearance-none border rounded w-full py-2 px-3 text-gray-700 leading-tight focus:outline-none focus:shadow-outline"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        className="mt-4 w-full bg-indigo-600 hover:bg-indigo-700 text-white font-bold py-2 px-4 rounded focus:outline-none focus:shadow-outline"
        onClick={[Function]}
      >
        Search
      </button>
    </div>
    <div />
  </div>
</div>
`;
